charts/dbildungs-iam/templates/dbildungs-iam-deployment-backend.yaml

```diff
@@ -1,45 +1,45 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: dbiam-deployment-backend
+  name: {{ include "dbildungs-iam.name" . }}-backend
   labels:
-    app.kubernetes.io/name: dbildungs-iam
+    {{- include "dbildungs-iam.labels" . | nindent 4 }}
 spec:
   selector:
     matchLabels:
-      layer: dbildungs-iam-backend
-  replicas: {{.Values.dbildungsIamReplications}}
+      {{- include "dbildungs-iam.backendLayerLabel" . | nindent 6 }}
+  replicas: {{ .Values.dbildungsIamReplications }}
   template:
     metadata:
-      name: dbildungs-iam
+      name: {{ include "dbildungs-iam.templateName" . }}
       labels:
-        layer: dbildungs-iam-backend
+        {{- include "dbildungs-iam.backendLayerLabel" . | nindent 8 }}
     spec:
       containers:
-        - name: dbildungs-iam-backend
-          image: {{.Values.dbildungsIamContainer}}
+        - name: {{ include "dbildungs-iam.name" . }}-backend
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
           imagePullPolicy: Always
           securityContext:
             allowPrivilegeEscalation: false
           ports:
```

Comment:
As we get the warning that containers should not run as root user, it may be best to block them from doing so by setting:
securityContext:  
  runAsNonRoot: true
However i am not sure if it is easier for developer to develop in a container in which they have root permissions, maybe we have to set them differently for dev and staging/prod stage? @dimapin